src/app/core/store/documents/documents.effects.ts

```diff
@@ -107,6 +110,32 @@ export class DocumentsEffects {
     })
   );
 
+  @Effect()
+  public createWithLink$: Observable<Action> = this.actions$.pipe(
+    ofType<DocumentsAction.CreateWithLink>(DocumentsActionType.CREATE_WITH_LINK),
+    mergeMap(action => {
+      const {document, otherDocumentId, linkTypeId, callback} = action.payload;
+      const documentDto = convertDocumentModelToDto(document);
+
+      return this.documentService.createDocument(documentDto).pipe(
+        map(dto => convertDocumentDtoToModel(dto)),
+        mergeMap(newDocument => {
+          const linkInstance: LinkInstance = {documentIds: [newDocument.id, otherDocumentId], linkTypeId};
+          const linkInstanceDto = convertLinkInstanceModelToDto(linkInstance);
+          return this.linkInstanceService.createLinkInstance(linkInstanceDto).pipe(
```

Comment:
Situation is quite different here. I need to scroll to new post-it after document and link is created. The other solution is to use callback after document creation, execute link action with callback and then scroll the post-it. But I'm afraid it will not work. Callback in these effects are executed before object is added to store, so post-it is not created yet. 

Solution 1: Use existing code.
Solution 2: Execute callbacks in effects after objects are added to store.
Solution 3: ??

Reply:
So what do you recommend?

Reply:
I can see that the callbacks for both document and link instance creation are not implemented the best way and they are executed before the success action. We should probably change that, however, it might break something so it does not make sense to do that in this PR. I think your solution is good for now. But we should try to avoid creating custom actions and action composition whenever possible.